vochain/app.go

```diff
@@ -506,8 +579,42 @@ func (app *BaseApplication) PrepareProposal(ctx context.Context,
 // Application SHOULD accept a prepared proposal passed via ProcessProposal, even if a part of the proposal
 // is invalid (e.g., an invalid transaction); the Application can ignore the invalid part of the prepared
 // proposal at block execution time. The logic in ProcessProposal MUST be deterministic.
```

Comment:
this comment says the opposite of what happens inside the func (returns ResponseProcessProposal_REJECT in case a transaction was invalid)

i trust your code, but then the comment needs updating (or an explanation why are you deviating from this SHOULD)

Reply:
I guess this comment is for the safety of the non-experneiced users. 

Here more details: https://github.com/cometbft/cometbft/blob/main/spec/abci/abci++_methods.md#processproposal

> Moreover, application implementors SHOULD always set ResponseProcessProposal.status to ACCEPT, unless they really know what the potential liveness implications of returning REJECT are.

So, if we accept any block proposed by a validator, we might run into the risk of including invalid transactions to blocks, which won't be executed since deliverTx() will fail, but the invalid tx will remain as part of the blockchain forever. Thus, a misbehaving validator could potentially fill the blockchain with garbage.

In the other side, if we reject the proposal, we force the validators to act honestly. The downside is that a block will take longer to be produced since a new validator needs to start over the transaction set for a new block.

I think, the way we do it should be safe since on prepare proposal, we are executing all tx against an ongoing state (reverted with Rollback() at the end) an on process proposal we do exactly the same. If the state is deterministic (which must be) the outcome should always be the same (either ACCEPT or REJECT).